tests/test_paperqa.py

```diff
@@ -1239,6 +1240,69 @@ def test_dois_resolve_to_correct_journals(doi_journals):
     assert details.journal == doi_journals["journal"]
 
 
+@pytest.mark.asyncio
+async def test_docdetails_merge_with_non_list_fields():
+    # Test merging a DocDetail that was republished
+    initial_date = datetime(2023, 1, 1)
+    doc1 = DocDetails(
+        citation="Citation 1",
+        publication_date=initial_date,
+        docname="Document 1",
+        dockey="key1",
+        other={"bibtex_source": "source1", "client_source": ["client1"]},
+    )
+
+    later_publication_date = initial_date + timedelta(weeks=13)
+    doc2 = DocDetails(
+        citation=doc1.citation,
+        publication_date=later_publication_date,
+        docname=doc1.docname,
+        dockey="key2",
+        other={"bibtex_source": ["source2"], "client_source": "client2"},
+    )
+
+    # Merge the two DocDetails instances
+    merged_doc = doc1 + doc2
+
+    # use set operations to check for containment
+    assert {"source1", "source2"}.issubset(merged_doc.other["bibtex_source"])
+    assert {"client1", "client2"}.issubset(merged_doc.other["client_source"])
+    # Check that the merged_doc is an instance of DocDetails
+    assert isinstance(merged_doc, DocDetails)
```

Comment:
```suggestion
    assert {"source1", "source2"}.issubset(merged_doc.other["bibtex_source"]), "Expected merge to keep both bibtex sources"
    assert {"client1", "client2"}.issubset(merged_doc.other["client_source"]), "Expected merge to keep both client sources"
    assert isinstance(merged_doc, DocDetails), "Merged doc should also be DocDetails"
```

Two comments:
- The code comments you put are mostly redundant to the code
- Use assertion messages to house your intent